numeric.c

```diff
@@ -1671,13 +1671,20 @@ rb_float_cmp(VALUE x, VALUE y)
 }
 
 /*
- * call-seq:
- *   float > real  ->  true or false
+ *  call-seq:
+ *    self > other -> true or false
+ *
+ *  Returns +true+ if +self+ is numerically greater than +other+:
  *
- * Returns +true+ if +float+ is greater than +real+.
+ *    2.0 > 1              # => true
+ *    2.0 > 1.0            # => true
+ *    2.0 > Rational(1, 2) # => true
+ *    2.0 > 2.0            # => false
  *
- * The result of <code>NaN > NaN</code> is undefined,
- * so an implementation-dependent value is returned.
+ *  <tt>Float::NAN > Float::NAN</tt> returns an implementation-dependent value.
+ *
+ *  Raises an exception if +other+ does not respond to method <tt>:></tt>
+ *  (including when +other+ is Complex).
```

Comment:
I don't think that is correct, at least not in all cases:

```ruby
Rational.undef_method(:>=)
Rational.undef_method(:>)
Rational.undef_method(:<=)
Rational.undef_method(:<)
Rational.undef_method(:<=>)
2.0 >= Rational(1, 2) # => true
2.0 > Rational(1, 2) # => true
2.0 <= Rational(1, 2) # => false
2.0 < Rational(1, 2) # => false
```

How it behaves is going to depend in some cases on `coerce`:

```ruby
class A; def coerce(j); [j, 1] end; end
A.new.respond_to?(:>) # => false
2.0 < A.new # => false
0.0 < A.new # => true
```

I would just drop this paragraph from the documentation from `>`, `>=`, `<`, and `<=`.

Reply:
Done.